torch/utils/data/decoder.py

```diff
@@ -0,0 +1,270 @@
+# This file takes partial of the implementation from NVIDIA's webdataset at here:
+# https://github.com/tmbdev/webdataset/blob/master/webdataset/autodecode.py
+
+import pickle
+import re
+import os
+
+import numpy as np
+import PIL
+import PIL.Image
+import json
+import tempfile
+import io
+
+
+################################################################
+# handle basic datatypes
+################################################################
+
+
+def basichandlers(key, data):
+
+    extension = re.sub(r".*[.]", "", key)
+
+    if extension in "txt text transcript":
+        return data.decode("utf-8")
+
+    if extension in "cls cls2 class count index inx id".split():
+        try:
+            return int(data)
+        except ValueError:
+            return None
+
+    if extension in "json jsn":
+        return json.loads(data)
+
+    if extension in "pyd pickle".split():
+        return pickle.loads(data)
+
+    if extension in "pth".split():
```

Comment:
`.pth` is not the best extension to standardize as PyTorch model extension: https://github.com/pytorch/pytorch/issues/14864

`.pt` is safer